assets/css/_header.scss

```diff
@@ -1,50 +1,103 @@
 .main-header {
-  align-items: center;
+  align-items: baseline;
   background-color: $lightest-gray;
   border-bottom: $base-border;
   display: flex;
+  flex-wrap: wrap;
   justify-content: space-between;
-  padding: $base-spacing;
+  padding: $small-spacing $base-spacing;
 
   @include media($small-screen) {
     padding: $small-spacing;
   }
 }
 
 .logo {
-  text-decoration: none;
-  font-family: $medium-font-family;
   color: $dark-blue;
+  font-family: $medium-font-family;
+  font-size: $medium-font-size;
+  margin-bottom: $base-spacing;
+  order: 2;
+  text-align: center;
+  text-decoration: none;
+
+  @include media($medium-screen-up) {
+    margin-bottom: 0;
+  }
 
   &:hover {
     text-decoration: none;
   }
 }
 
-.navigation {
-  display: flex;
-
-  li {
-    margin-right: $small-spacing;
+.settings-gravatar {
+  order: 1;
 
-    &:last-child {
-      margin-right: 0;
-    }
+  @include media($medium-screen-up) {
+    order: 3;
   }
 }
 
 input[type="text"].header-search-input {
-  @include size(280px $button-small-height);
+  @include size(100% $button-small-height);
   font-size: $small-font-size;
   margin-bottom: 0;
+  text-align: center;
 
-  @include media($small-screen) {
-    width: 100px
+  @include media($medium-screen-up) {
+    @include size(50% null);
+    margin: 0 auto;
   }
 }
 
 .current-user-gravatar {
-  border-radius: $base-border-radius;
-  vertical-align: middle;
+  border-radius: 50%;
   height: $button-small-height;
+  margin-top: -$tiny-spacing;
+  vertical-align: middle;
+
+  @include media($medium-screen-up) {
+    margin-right: $small-spacing;
+  }
+}
+
+.header-search-form {
+  flex: 1 0 100%;
+  order: 4;
+  position: relative;
+
+  @include media($medium-screen-up) {
+    flex-basis: 18.75rem;
+    order: 2;
+  }
+}
+
+.header-search-form::before {
+  @include position(absolute, 28% null null 20%);
```

Comment:
I feel like this lil guy should be locked to the left side of the input (with a slight offset so it looks like it's inside like now). It was a slightly jarring for it to move around on resize.